crates/subspace-service/src/sync_from_dsn/import_blocks.rs

```diff
@@ -93,25 +93,42 @@ where
         trace!(
             target: LOG_TARGET,
             %segment_index,
-            last_archived_block_number,
+            last_archived_maybe_partial_block_number,
             last_archived_block_partial,
             "Checking segment header"
         );
 
-        let last_archived_block_number = NumberFor::<Block>::from(last_archived_block_number);
+        let (last_partial_archived_block_number, last_full_archived_block_number) =
+            if last_archived_block_partial {
+                // The genesis block is always fully reconstructed, so we can saturating_sub here
+                (
+                    Some(NumberFor::<Block>::from(
+                        last_archived_maybe_partial_block_number,
+                    )),
+                    NumberFor::<Block>::from(last_archived_maybe_partial_block_number)
+                        .saturating_sub(1u32.into()),
+                )
+            } else {
+                (
+                    None,
+                    NumberFor::<Block>::from(last_archived_maybe_partial_block_number),
+                )
+            };
```

Comment:
I would simplify this, it is unnecessarily complex:
```suggestion
        let last_full_archived_block_number = if last_archived_block_partial {
            // The genesis block is always fully reconstructed, so we can saturating_sub here
            NumberFor::<Block>::from(last_archived_maybe_partial_block_number)
                .saturating_sub(1u32.into())
        } else {
            NumberFor::<Block>::from(last_archived_maybe_partial_block_number)
        };
```

`last_partial_archived_block_number` is pointless because we already have `last_archived_maybe_partial_block_number` stored above and condition below didn't really need to change beyond variable name change.